### PR TITLE
Bind the maps setup event on viewDidInsert and mapOptions modification.

### PR DESCRIPTION
**Pull request by erkie**
Makes it possible to redraw the map by changing the `mapOptions` parameter.

I also did some formatting in this branch. Removed spaces between `functionName : function` and changed some instances of `var` to `let`. This might be a bit controversial so please feel free to comment on it and I will change to whatever you feel is correct. If you want I can also split this into two commits, one for formatting and one for the actual fix.

@SamvelRaja please have a look! Also, is it OK that I'm committing and PR'ing like this? Or would you prefer another way of working?

### CODE REVIEW

File: addon/components/google-maps-addon.js [hunk 1]
```diff
@@ -1,19 +1,22 @@
 import Ember from 'ember';
 import Helpers from '../helpers';
+
 export default Ember.Component.extend({
-  didInsertElement: function() {
-    var mapElement;
-    //Initializing the options into the context
+  setupMap: Ember.observer('mapOptions', Ember.on('didInsertElement', function() {
```
Comment (SamvelRaja) [inline addon/components/google-maps-addon.js:5]: Nice catch :+1:

File: addon/helpers.js [hunk 2]
```diff
@@ -61,8 +60,8 @@ export default{
   @usage
     To initialize the `mouseevents` to the `mapElement`
   **/
-  initializeMouseEventCallbacks : function(context) {
-    let mapOptions = context.get('mapOptions') || context.get('MapOptions');
```
Comment (SamvelRaja) [inline addon/helpers.js:65]: so we are removing the `MapOptions` support then it must be a major version change.

**Comment by SamvelRaja** [comment]
@erkie **Creating branches** and merging would be nice, and you go ahead and merge it if they are normal fixes.
For features and enhancements its better to have a conversation :+1:

**Comment by SamvelRaja** [comment]
Also, these styles contradictions will gets fixed once this proposal has been reinforced.

**Comment by erkie** [comment]
Thanks!

wrjpk6o

**Pull request merged**